python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -737,7 +737,7 @@ def get_nodes_for(node_types: Dict[NodeType, NodeTypeConfigDict],
 def _utilization_score(node_resources: ResourceDict,
                        resources: List[ResourceDict]) -> Optional[float]:
     remaining = copy.deepcopy(node_resources)
-    is_gpu_node = "GPU" in node_resources
+    is_gpu_node = "GPU" in node_resources and node_resources["GPU"] != 0
```

Comment:
How about ">"?

Reply:
Sounds good.

Though, for peak pedantry (and eventual introduction of type-checking), I will have to change the type values of this dict from numbers.Number to numbers.Real